data/creaturescripts/scripts/boss.lua

```diff
@@ -0,0 +1,229 @@
+	db.asyncStoreQuery('SELECT `pid`, `sid` FROM `player_rewards` WHERE player_id = ' .. playerGuid .. ' ORDER BY `sid` ASC;', 
+		function(query)
+			local lastReward = 0
+			local lastStoreId   
+			if query then             
+				repeat
+					local sid = result.getDataInt(query, 'sid')
+					local pid = result.getDataInt(query, 'pid')
+
+					if pid < 100 then
+						lastReward = pid
+					end
+					lastStoreId = sid
+				until not result.next(query)
+			end
+
+			local buffer = {'INSERT INTO `player_rewards` (`player_id`, `pid`, `sid`, `itemtype`, `count`, `attributes`) VALUES'}
+
+			--reward bag
+			local info = {
+				playerGuid = playerGuid,
+				running = lastStoreId or 100
+			}
+
+			local bag = Game.createItem(ITEM_REWARD_CONTAINER)
+			bag:setAttribute(ITEM_ATTRIBUTE_DATE, timestamp)
+
+			if itemList then
+				for _, item in ipairs(itemList) do
+					bag:addItem(item[1], item[2])
+				end
+			end
+
+			local total = insertItems(buffer, info, lastReward + 1, {bag})
+			table.insert(buffer, ";")
+
+			if total ~= 0 then
+				db.query(table.concat(buffer))
+			end
+		end
+	)
+end
+
+local function getPlayerStats(bossId, playerGuid, autocreate)
+	local ret = globalBosses[bossId][playerGuid]
+	if not ret and autocreate then
+		ret = {
+			bossId = bossId, 
+			damageIn = 0, -- damage taken from the boss
+			healing = 0, -- healing (other players) done
+		}
+		globalBosses[bossId][playerGuid] = ret
+		return ret
+	end
+	return ret
+end
+
+function onDeath(creature, corpse, killer, mostDamageKiller, lastHitUnjustified, mostDamageUnjustified)
+	local monsterType = creature:getType()
+	if monsterType:isRewardBoss() then -- Make sure it is a boss
+		local bossId = creature:getId()
+		local timestamp = os.time()
+
+		local totalDamageOut, totalDamageIn, totalHealing = 0.1, 0.1, 0.1 -- avoid dividing by zero
+
+		local scores = {}
+		local info = globalBosses[bossId]
+		local damageMap = creature:getDamageMap()
+
+		for guid, stats in pairs(info) do
+			local player = Player(stats.playerId)
+			local part = damageMap[stats.playerId]
+			local damageOut, damageIn, healing = (stats.damageOut or 0) + (part and part.total or 0), stats.damageIn or 0, stats.healing or 0
+
+			totalDamageOut = totalDamageOut + damageOut
+			totalDamageIn = totalDamageIn + damageIn
+			totalHealing = totalHealing + healing
+
+			table.insert(scores, {
+				player = player, 
+				guid = guid,
+				damageOut = damageOut, 
+				damageIn = damageIn,
+				healing = healing,
+			})            
+		end
+
+		local participants = 0
+		for _, con in ipairs(scores) do
+			local score = (con.damageOut / totalDamageOut) + (con.damageIn / totalDamageIn) + (con.healing / totalHealing)
+			con.score = score / 3 -- normalize to 0-1
+			if score ~= 0 then
+				participants = participants + 1
+			end
+		end
+		table.sort(scores, function(a, b) return a.score > b.score end)
+
+		local expectedScore = 1 / participants
+
+		for _, con in ipairs(scores) do
```

Comment:
The inner `score` should be named `value` then since if you have a list of `scores` , every item is a `score` by itself.